lib/metrics/src/serve.rs

```diff
@@ -113,7 +113,7 @@ impl fmt::Display for ServeError {
             f,
             "{}: {}",
             self.description(),
-            self.cause().expect("ServeError must have cause")
+            self.source().expect("ServeError must have cause")
```

Comment:
Take it or leave it: maybe it's worth changing the `expect` message to
```suggestion
            self.source().expect("ServeError must have source")
```
but it's not super important.